dask/utils.py

```diff
@@ -647,6 +648,116 @@ def __doc__(self):
             return "Single Dispatch for %s" % self.__name__
 
 
+class DaskBackendEntrypoint:
```

Comment:
Minor point: `dask/utils.py` doesn't feel like the right place for the code added here. It seems more important that other utilities here. Maybe `dask/base.py` or a new `dask/backends.py` module instead?

Reply:
I agree that utils feels off. I moved this to a new `dask/backends.py` module for now.